wal/wal.go

```diff
@@ -41,6 +41,10 @@ const (
 
 	// the owner can make/remove files inside the directory
 	privateDirMode = 0700
+
+	// the expected size of each wal segment file.
+	// the actual size might be bigger than it.
+	segmentSizeBytes = 8 * 1000 * 1000 // 8MB
```

Comment:
Why is it 8MB? I mean, it sounds like a number -- what brought you to that value? Is it based on an approximate number of entries?

And then, what happens if one entry, say, has a value which is really huge? Will we sit there cutting a lot of files?